.github/workflows/build-release.yml

```diff
@@ -0,0 +1,193 @@
+---
+name: "Build Release"
+on:
+  workflow_dispatch:
+    inputs:
+      releaseBranch:
+        description: "The branch on which a release is based"
+        required: false
+        default: master
+      releaseVersion:
+        description: "Version to be used as tag and release"
+        required: true
+        default: X.Y.Z
+      developmentVersion:
+        description: 'Post-release Development version. Should be "(releaseVersion + 1)-SNAPSHOT"'
+        required: true
+        default: X.Y.Z-SNAPSHOT
+      awsRegion:
+        description: "AWS Region to upload artifacts to"
+        required: true
+        default: us-east-1
+
+permissions:
+  id-token: write # This is required for requesting the AWS IAM OIDC JWT
+  contents: write # This is required for actions/checkout
+
+env:
+  # AWS Code Artifact Repository
+  CA_REPOSITORY: bfd-mgmt
+  CA_DOMAIN: bfd-mgmt
+  AWS_REGION: ${{ inputs.awsRegion }}
+
+jobs:
+  mvn-release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Validate Inputs
+        run: |
+          echo "Validating inputs to ensure they conform to expected formats..."
+          echo "${{ inputs.releaseVersion }}" | grep -P '^\d+\.\d+\.\d+$'
+          echo "${{ inputs.developmentVersion }}" | grep -P '^\d+\.\d+\.\d+-SNAPSHOT$'
+
+      - name: "Generate an App Token"
+        id: generate_token
+        uses: actions/create-github-app-token@v1
+        with:
+          app-id: ${{ secrets.BFD_RELEASE_APP_ID }}
+          private-key: ${{ secrets.BFD_RELEASE_APP_KEY }}
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          role-to-assume: ${{ secrets.GHA_AWS_IAM_ROLE_ARN }}
+          role-session-name: build-release
+          aws-region: ${{ env.AWS_REGION }}
+
+      - name: Login to ECR
+        uses: aws-actions/amazon-ecr-login@v1
+        with:
+          mask-password: "true"
+
+      - name: Checkout
+        if: github.event_name == 'workflow_dispatch'
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          ref: ${{ inputs.releaseBranch }}
+          token: ${{ steps.generate_token.outputs.token }}
+
+      - name: Setup JDK
+        uses: actions/setup-java@v3
+        with:
+          java-version: "17"
+          distribution: corretto
+
+      - name: Configure the git user
+        run: |
+          git config --global user.email "actions@github.com"
+          git config --global user.name "GitHub Actions"
+
+      - name: Generate maven toolchain config
+        run: |
+          cat << EOF > ~/.m2/toolchains.xml
+          <toolchains>
+            <toolchain>
+              <type>jdk</type>
+              <provides>
+                <version>17</version>
```

Comment:
Just a note this will need to be updated to v21 soon ™️

Reply:
Necromancing this comment 💀. With #2066 having landed as of yesterday, I just want to make things a _little_ easier to reason about. Let's be sure that the target branch is up-to-date with the trunk and that **this** branch is up-to-date with the target.

To that end, can you open a draft PR for BFD-3078 and... make it so?